frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -276,22 +282,38 @@ export const HorizontalNav = React.memo((props: HorizontalNavProps) => {
     [navTabExtensions, objReference],
   );
 
-  const dynamicPluginPages = React.useMemo(
-    () =>
-      navTabExtentionsResolved
-        ? dynamicNavTabExtensions
-            .filter(
-              (tab) =>
-                referenceForExtensionModel(tab.properties.model as ExtensionK8sGroupModel) ===
-                objReference,
-            )
-            .map((tab) => ({
-              ...tab.properties.page,
-              component: tab.properties.component,
-            }))
-        : [],
-    [dynamicNavTabExtensions, navTabExtentionsResolved, objReference],
-  );
+  const dynamicPluginPages = React.useMemo(() => {
+    const resolvedResourceNavTab = navTabResourceExtentionsResolved
+      ? dynamicResourceNavTabExtensions
+          .filter(
+            (tab) =>
+              referenceForExtensionModel(tab.properties.model as ExtensionK8sGroupModel) ===
+              objReference,
+          )
+          .map((tab) => ({
+            ...tab.properties.page,
+            component: tab.properties.component,
+          }))
+      : [];
+
+    const resolvedNavTab = navExtentionsResolved
+      ? dynamicTabExtensions
+          .filter((tab) => tab.properties.contextId === contextId)
+          .map((tab) => ({
+            ...tab.properties.page,
+            component: tab.properties.component,
+          }))
+      : [];
+
+    return [...resolvedResourceNavTab, ...resolvedNavTab];
```

Comment:
We need to ensure there is stable ordering if multiple plugins extend the same horizontal nav.

Reply:
I added a string based sorting for now. Until we include the `insertBefore` `insertAfter`

While checking the `insertBefore` `insertAfter` implementation seems other extensions are using  [order-extensions.ts](https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/utils/order-extensions.ts) which requires an id. Should we add the `id` property now, so adding the sorting later will be easier?